### PR TITLE
Swap Py_BuildValue with pg_tuple_from_values_int

### PR DESCRIPTION
**Pull request by MightyJosip**
Connected: #3530,  #3424, #3401

### CODE REVIEW

**Comment by itzpr3d4t0r** [comment]
Approved!

File: src_c/color.c [hunk 1]
```diff
@@ -1841,10 +1841,10 @@ _color_slice(register pgColorObject *a, register Py_ssize_t ilow,
         return Py_BuildValue("(iiii)", c1, c2, c3, c4);
     }
     else if (len == 3) {
-        return Py_BuildValue("(iii)", c1, c2, c3);
+        return pg_tuple_triple_from_values_int(c1, c2, c3);
     }
     else if (len == 2) {
-        return Py_BuildValue("(ii)", c1, c2);
+        return pg_tuple_couple_from_values_int(c1, c2);
```
Comment (itzpr3d4t0r) [inline src_c/color.c:1847]: These should be cast to ints

**Review by ankith26 — COMMENTED** [review]
Thanks for the PR! The lint CI is failing so we need a round of `setup.py format` here. Also some places probably need explicit `int` casts, because otherwise it would be implicit.

File: src_c/color.c (hunk 1 above)
Comment (Starbuck5) [reply src_c/color.c:1847]: Yes-- this is why the appveyor builds are failing.

**Comment by Starbuck5** [comment]
I did some performance testing on the change of method itself.

```py
import pygame
import time

a = time.time()
for i in range(10):
    pygame.transform.test()
print(time.time() - a)
```

```c
static PyObject*
testtest(PyObject *self, PyObject *args)
{
    int x = 15;
    int y = 772;
    int z = 12;
    PyObject* result;

    for (int i=0; i<500000; i++) {
        //result = Py_BuildValue("(ii)", x, y);
        //result = pg_tuple_couple_from_values_int(x, y);
        //result = Py_BuildValue("(iii)", x, y, z);
        result = pg_tuple_triple_from_values_int(x, y, z);
    }

    Py_RETURN_NONE;
}

...
{"test", (PyCFunction)testtest, METH_NOARGS, NULL},
```

**Results**
```
# 500k*10 tuple II 
# BuildValue: 0.33 seconds
# custom: 0.22 seconds

# 500k*10 tuple III
# BuildValue: 0.35 seconds
# custom: 0.24 seconds
```

So 5 million calls translates into a savings of only 0.11 seconds. Therefore it would take around 45 million calls to something using this to notice 1 second of runtime improvement. This is very much a micro-optimization. But I don't see it adding maintenance burden for us, so I'm pro.

**Comment by itzpr3d4t0r** [comment]
I think your results are a bit incorrect, firstly because the C code has a memory leak ( you didn't DECREF result) and second because by doing 
```Python
a = time.time()
for i in range(10):
    pygame.transform.test()
print(time.time() - a)
```
You are including the creation time of the range object, the looping and access to the module and in reality when this function is called internally in C you don't do all this stuff, so i came up with a slightly different measuring strategy:

- First i fixed the memry leak which was actually changing the results apparently
- Second i changed the Python code to the much more precise timeit, excluding
- I also made the measurements 5 times in order to check whether the first call add extra overhead:
```Python
from pygame.transform import test
from timeit import timeit

for _ in range(5):
    print(timeit(test, number=10))
```
And got these results:

```
TRIPLE

old
0.24411700000018755
0.2521020999997745
0.24876799999992727
0.2405932000001485
0.2455758999999489

new
0.12007300000004761
0.12075110000023415
0.12277529999983017
0.11842769999975644
0.12147270000014032

=====
DOUBLE

old
0.2224092000001292
0.22611889999961932
0.2230251999999382
0.22691179999992528
0.22417660000019168

new
0.11334439999973256
0.1101702999999361
0.11015529999986029
0.11212010000008377
0.11333589999958349
```

So from my measurements i get roughly 96% faster for the double and 106% faster for the triple.

**Comment by Starbuck5** [comment]
@MightyJosip Lets move this forward.

Ankith and itzpr pointed out small things that need fixing. (casts + format run)

Once you do that the builds will probably start failing because of a sphinx issue, you'll need to rebase to get the fix for that.

**Comment by Starbuck5** [comment]
MightyJosip on discord:

> About your comment, I am gonna do it next month, I don't have time now, next 3 weeks should hopefully be the last weeks of my university :hype: (not counting writing the thesis), so I will have more time

**Pull request closed without merging**